cmd/buildah/run.go

```diff
@@ -178,5 +179,28 @@ func runCmd(c *cobra.Command, args []string, iopts runInputOptions) error {
 		conditionallyAddHistory(builder, c, "%s %s", shell, strings.Join(args, " "))
 		return builder.Save()
 	}
+	// unlock if any locked files from this RUN statement
+	for _, path := range lockedTargets {
+		_, err := os.Stat(path)
+		if err != nil {
+			// Lockfile not found this might be a problem,
+			// since LockedTargets must contain list of all locked files
+			// don't break here since we need to unlock other files but
+			// log so user can take a look
+			logrus.Errorf("Lockfile %q was expected here, stat failed with %v", path, err)
+			continue
+		}
+		lockfile, err := lockfile.GetLockfile(path)
+		if err != nil {
+			// unable to get lockfile
+			// lets log error and continue
+			// unlocking other files
+			logrus.Error(err)
+			continue
+		}
+		if lockfile.Locked() {
```

Comment:
At this point it should be an error if the file isn't locked, because we either unlocked it somewhere else by mistake, lost the lock for some reason, or `GetVolumes()` misled us.

Reply:
Producing an error here for such anomaly cases.